documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler.adoc

```diff
@@ -48,7 +48,38 @@ Refer to xref:pg-server-http-handler-use[this section] for more information abou
 You should write your own leaf ``Handler``s to implement your web application logic.
 Refer to xref:pg-server-http-handler-impl[this section] for more information about how to write your own ``Handler``s.
 
-// TODO: document ScopedHandler? Is this really necessary or just an implementation detail that application will never worry about?
+A leaf `Handler` may be declared as non-blocking (by extending `Handler.Abstract.NonBlocking`) or as blocking (by extending `Handler.Abstract`), to allow interaction with the xref:pg-arch-threads[Jetty threading architecture] for more efficient thread and CPU utilization during the request/response processing.
```

Comment:
It is not just leaf handlers.   All handlers in a calling chain must be non-blocking

```suggestion
A `Handler` may be declared as non-blocking (by extending `Handler.Abstract.NonBlocking`) or as blocking (by extending `Handler.Abstract`), to allow interaction with the xref:pg-arch-threads[Jetty threading architecture] for more efficient thread and CPU utilization during the request/response processing.
```

Reply:
@gregw yes, but `Handler.Wrapper` inherit the non-blocking-ness from their children, so it's important to define it only for leaf `Handler`s -- the others have good defaults.

Reply:
But you explain that below. So why start with something incorrect?